spec/app/domain/authentication/authn-oidc/update_input_with_username_from_id_token_spec.rb

```diff
@@ -42,23 +45,35 @@ def request_body(request)
   end
 
   let(:authenticate_id_token_request) do
-    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"alice\"}")
+    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"alice\"}", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_missing_id_token_username_field) do
-    mock_authenticate_oidc_request(request_body_data: "id_token={}")
+    mock_authenticate_oidc_request(request_body_data: "id_token={}", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_empty_id_token_username_field) do
-    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"\"}")
+    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"\"}", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_missing_id_token_field) do
-    mock_authenticate_oidc_request(request_body_data: "some_key=some_value")
+    mock_authenticate_oidc_request(request_body_data: "some_key=some_value", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_empty_id_token_field) do
-    mock_authenticate_oidc_request(request_body_data: "id_token=")
+    mock_authenticate_oidc_request(request_body_data: "id_token=", request_headers: {"CONTENT_TYPE" => "text/plain"})
+  end
+
+  let(:authenticate_id_token_request_id_token_in_header_field) do
+    mock_authenticate_oidc_request(request_body_data: "id_token=", request_headers: {"HTTP_AUTHORIZATION" => "Bearer {\"id_token_username_field\":\"alice\"}"})
```

Comment:
I think we are missing tests where the `id_token` key is not present in the body at all, that the body is totally empty

Reply:
added

Reply:
Doesn't the `Bearer` value come through as a Base64 encoded JWT?

Reply:
Just saw the comment below.  Maybe one of these is Base64 encoded to ensure the typical flow is tested?

Reply:
The token is base64 encoded, we implemented same test as the `id_token` present in the body, will create integration test for Conjur cloud UI in different story

Reply:
Please add at least a unit test here as well. I'm not comfortable adding functionality without tests to back it up.  The Core team will be on the line for supporting this work.

Reply:
added integration test

Reply:
Thanks!